### PR TITLE
THRIFT-5650: Implement UUID in Go compiler

### PR DESCRIPTION
**Pull request by fishy**
Client: go

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [ ] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

**Comment by fishy** [comment]
hmm go-swift cross-test failed.

@Jimexist do you know how to access the detailed log from github actions?

**Comment by fishy** [comment]
I found the url to the logs uploaded by failed cross-test by enabling debug logging, but I don't have access to it.

**Comment by kinoroy** [comment]
go-swift is failing because Go cross test server does not implement testUuid() method. This causes an error when the Swift client calls it as part of cross:
```
TApplicationError (1): Unknown Method
Message: Unknown function testUuid
```

**Comment by fishy** [comment]
@kinoroy thanks that's easy to fix.

just out of curiosity, did you run the crosstests locally or did you get that from the logs from github actions?

**Comment by kinoroy** [comment]
@fishy Sorry I forgot to mentioned, you can get the logs from GitHub on the job page once its complete just click "cross-test-log" to download :) 
Screen Shot 2022-10-22 at 5 00 26 PM

File: lib/go/thrift/json_protocol.go [hunk 1]
```diff
@@ -524,6 +524,8 @@ func (p *TJSONProtocol) TypeIdToString(fieldType TType) (string, error) {
 		return "set", nil
 	case LIST:
 		return "lst", nil
+	case UUID:
+		return "uid", nil
```
Comment (fishy) [inline lib/go/thrift/json_protocol.go:528]: apparently we don't have a written spec for TJSONProtocol, so I followed this part from the implementation of java library.

**Comment by fishy** [comment]
It looks like rust is also using the wrong TType value for uuid so the rs-go tests are failing.

**Review by Jimexist — APPROVED** [review]

**Pull request merged**